File: 프로그래머스/lv3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0
    const nodes = Array.from({length:computers.length},()=>0);
    const go = (i)=>{
        nodes[i] = 1;
        const info = computers[i];
        for(let j = 0;j<info.length;j++){
            if(info[j]===1&&nodes[j]===0) go(j);
        }
    }
    for(let i = 0;i<nodes.length;i++){
        if(nodes[i]===0) {
            answer +=1;    
            go(i)
        };
    }

    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    isVisited = [False for _ in range(n)]
    
    cnt = 0
    def dfs(c):
        isVisited[c] = True;
        
        for next,isLinked in enumerate(computers[c]):
            if(not isVisited[next] and isLinked):
                dfs(next)
        
    for i in range (n):
        if not isVisited[i]:
            dfs(i)
            cnt+=1
            
    
    return cnt